File: app/src/main/java/app/quranhub/ui/mushaf/presenter/Mus7fPresenter.java
package app.quranhub.ui.mushaf.presenter;

import app.quranhub.ui.base.BasePresenter;
import app.quranhub.ui.base.BaseView;
import app.quranhub.data.local.entity.TranslationBook;

public interface Mus7fPresenter<T extends BaseView> extends BasePresenter<T> {

    void getQuranPageInfo(int curentPage);

    void setNightMode(boolean nightMode);

    boolean getNightMode();

    boolean toggleNightMode();

    void getAyaTafseer(int ayaId);

    void getCurrentTafseerBook(String currentTafsserId);

    void onGetTafsserBook(TranslationBook book);

    void getSurasInPage();

    void checkAyaHasRecorder(int id);

    void saveRecorderPath(int ayaId, String recorderPath);

    void deleteAyaVoiceRecorder(int ayaId);

    void getSuraNumofVerses();

    void getFromAyaPage(int fromAya);

    void getNotificationAya(int ayaId);

}

File: app/src/main/java/app/quranhub/ui/mushaf/presenter/Mus7fPresenter.kt
package app.quranhub.ui.mushaf.presenter

import app.quranhub.data.local.entity.TranslationBook
import app.quranhub.ui.base.BasePresenter
import app.quranhub.ui.base.BaseView

interface Mus7fPresenter<T : BaseView> : BasePresenter<T> {
    fun getQuranPageInfo(curentPage: Int)
    fun setNightMode(nightMode: Boolean)
    val nightMode: Boolean
    val quranPageZoomScaleFactor: Float
    fun toggleNightMode(): Boolean
    fun getAyaTafseer(ayaId: Int)
    fun getCurrentTafseerBook(currentTafsserId: String)
    fun onGetTafsserBook(book: TranslationBook)
    fun getSurasInPage()
    fun checkAyaHasRecorder(id: Int)
    fun saveRecorderPath(ayaId: Int, recorderPath: String)
    fun deleteAyaVoiceRecorder(ayaId: Int)
    fun getSuraNumofVerses()
    fun getFromAyaPage(fromAya: Int)
    fun getNotificationAya(ayaId: Int)
}
